### PR TITLE
Adpcm pipeline

### PR DESCRIPTION
**Pull request by MosesofEgypt**
Changes to sound_decompilation.py should NOT be included in merge.

### CODE REVIEW

File: sounds/src/shared.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+typedef uint16_t uint16;
+typedef uint32_t uint32;
+typedef uint64_t uint64;
+
+#define MAX_AUDIO_CHANNEL_COUNT 7
+
+#define RELEASE_PY_BUFFER_ARRAY(bufs)\
+    for (int py_buffer_release_index = 0; py_buffer_release_index < (sizeof(bufs) / sizeof(bufs[0])); py_buffer_release_index++) {\
+        PyBuffer_Release(&bufs[py_buffer_release_index]);\
+    }
```
Comment (ShellyHerself) [inline sounds/src/shared.h]: Missing newline!!!!

**Review by ShellyHerself — CHANGES_REQUESTED** [review]
Please undo the changes in decompilation if you do not want them merged.

File: sounds/src/shared.h (hunk 1 above)
Comment (MosesofEgypt) [reply sounds/src/shared.h]: Fixed

File: setup.py [hunk 2]
```diff
@@ -19,7 +51,7 @@
     long_desc = 'Could not read long description from readme.'
 
 
-setup(
+setup_kwargs = dict(
     name='reclaimer',
     description='A libray of SupyrStruct structures and objects for \
```
Comment (ShellyHerself) [inline setup.py]:
```suggestion
    description='A libray of SupyrStruct structures and objects for '
```

File: setup.py [hunk 3]
```diff
@@ -19,7 +51,7 @@
     long_desc = 'Could not read long description from readme.'
 
 
-setup(
+setup_kwargs = dict(
     name='reclaimer',
     description='A libray of SupyrStruct structures and objects for \
 games built with the Blam engine',
```
Comment (ShellyHerself) [inline setup.py]:
```suggestion
        'games built with the Blam engine',
```

**Review by ShellyHerself — APPROVED** [review]

**Pull request merged**